xarray/core/indexes.py

```diff
@@ -196,6 +196,50 @@ def create_variables(
         else:
             return {}
 
+    def should_add_coord_to_array(
+        self,
+        name: Hashable,
+        var: Variable,
+        dims: set[Hashable],
+    ) -> bool:
+        """Define whether or not an index coordinate variable should be added in
+        a new DataArray.
+
+        This method is called repeatedly for each Variable associated with
+        this index when creating a new DataArray (via its constructor or from a
+        Dataset) or updating an existing one.
```

Comment:
Can you define what "associated to this index" means more precise?

Reply:
I added a sentence just after this one.